### PR TITLE
Refactor/remove xline client

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:
In the previous releases, for testing purposes, we conducted functional tests on the lock server's RPC using the etcd client. However, starting from v0.4.1, since the xline-client has been preliminarily completed, we can re-implement some of the testing scenarios based on the xline-client.

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: xline-test-utils/src/bin/validation_lock_client.rs [hunk 1]
```diff
@@ -1,7 +1,11 @@
 //! this binary is only used for the validation of lock service
 
 use clap::{Parser, Subcommand};
-use etcd_client::Client;
```
Comment (iGxnon) [inline xline-test-utils/src/bin/validation_lock_client.rs:4]: could we remove etcd_client dependency in Cargo.toml?

Comment (Phoenix500526) [reply xline-test-utils/src/bin/validation_lock_client.rs:4]: Removed

File: xline/src/bin/restore_client.rs [hunk 2]
```diff
@@ -1,19 +0,0 @@
-use std::path::PathBuf;
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply xline/src/bin/restore_client.rs:1]: This restore client is not used in xline.

**Comment by bsbds** [comment]
I think we keep the original etcd-client, because if we only use xline-client, the servers in `xline/src/server` will not be tested in integration tests. Maybe add a trait so that we can test using both clients?

**Comment by Phoenix500526** [comment]
> I think we keep the original etcd-client, because if we only use xline-client, the servers in `xline/src/server` will not be tested in integration tests. Maybe add a trait so that we can test using both clients?

The integration test only cares about the correctness of the xline server. Therefore, it doesn't matter what exactly the client is. The compatibility with etcd client can be covered in the validation test.

**Review by bsbds — APPROVED** [review]

File: scripts/quick_start.sh [hunk 3]
```diff
@@ -59,7 +59,7 @@ run_container() {
     for ((i = 1; i <= ${size}; i++)); do
         docker run -d -it --rm --name=node${i} --net=xline_net --ip=${SERVERS[$i]} --cap-add=NET_ADMIN --cpu-shares=1024 -m=512M -v ${DIR}:/mnt ${image} bash &
     done
-    docker run -d -it --rm --name=node4 --net=xline_net --ip=${SERVERS[0]} --cap-add=NET_ADMIN --cpu-shares=1024 -m=512M -v ${DIR}:/mnt gcr.io/etcd-development/etcd:v3.5.5 bash &
+    docker run -d -it --rm --name=node4 --net=xline_net --ip=${SERVERS[0]} --cap-add=NET_ADMIN --cpu-shares=1024 -m=512M -v ${DIR}:/mnt ghcr.io/xline-kv/etcdctl:v3.5.9 bash &
```
Comment (iGxnon) [inline scripts/quick_start.sh:62]: Do we have this image available?

Comment (Phoenix500526) [reply scripts/quick_start.sh:62]: Yes, FYI: https://github.com/orgs/xline-kv/packages/container/package/etcdctl

**Comment by Phoenix500526** [comment]
We don't need to construct two inner clients in a `BenchClient`. It's easy to solve this problem, so I leave a good first issue(#486 ) to track it.

**Review by iGxnon — APPROVED** [review]

**Pull request merged**